src/main/java/net/dv8tion/jda/internal/interactions/InteractionImpl.java

```diff
@@ -104,6 +111,14 @@ public InteractionImpl(JDAImpl jda, DataObject data)
             }
             this.user = user;
         }
+
+        DataArray entitlementsArray = data.getArray("entitlements");
+        entitlements = new ArrayList<>();
```

Comment:
This should be like `this.entitlements = data.getArray("entitlements").stream(...)...collect(Collectors.toList());`